internal/manifests/collector/service.go

```diff
@@ -53,24 +53,23 @@ func HeadlessService(params manifests.Params) *corev1.Service {
 	h.Annotations = annotations
 
 	h.Spec.ClusterIP = "None"
-	return h
+	return h, nil
 }
 
-func MonitoringService(params manifests.Params) *corev1.Service {
+func MonitoringService(params manifests.Params) (*corev1.Service, error) {
 	name := naming.MonitoringService(params.OtelCol.Name)
 	labels := Labels(params.OtelCol, name, []string{})
 
 	c, err := adapters.ConfigFromString(params.OtelCol.Spec.Config)
 	// TODO: Update this to properly return an error https://github.com/open-telemetry/opentelemetry-operator/issues/1972
 	if err != nil {
 		params.Log.Error(err, "couldn't extract the configuration")
-		return nil
+		return nil, err
 	}
 
 	metricsPort, err := adapters.ConfigToMetricsPort(params.Log, c)
```

Comment:
note to other reviewers, we no longer return an error if we use the default metrics port.